src/parallel/Splitter.cc

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022, Antti Hyvarinen <antti.hyvarinen@gmail.com>
+ * Copyright (c) 2022, Seyedmasoud Asadzadeh <seyedmasoud.asadzadeh@usi.ch>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "Splitter.h"
+
+
+void Splitter::set_solver_branch(std::string & solver_branch)
+{
+    solverBranch.clear();
+    solver_branch.erase(std::remove(solver_branch.begin(), solver_branch.end(), ' '), solver_branch.end());
+    std::string const delimiter = "," ;
+    size_t beg, pos = 0;
+    uint16_t counter = 0;
+    uint16_t temp = 0;
```

Comment:
I don't think they're literally guaranteed, and even in practice I suppose it's possible that somebody wants to split an instance to more than 65535 parts, or that there is a branch longer than that.

I also think that in this case, running on a CPU that does natively 32-bit operations and since memory usage or cache misses is obviously not an issue here, it's actually faster to use the 32-bit integers.

There's some discussion for instance in here: https://stackoverflow.com/questions/163254/on-32-bit-cpus-is-an-integer-type-more-efficient-than-a-short-type

In summary, there's no benefit from using 16-bit integers, and it might in some cases which we haven't thought about yet cause overflow bugs.